compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -89,21 +87,22 @@ pub(crate) fn eval_to_valtree<'tcx>(
 #[instrument(skip(tcx), level = "debug")]
 pub(crate) fn try_destructure_mir_constant<'tcx>(
     tcx: TyCtxt<'tcx>,
-    param_env: ty::ParamEnv<'tcx>,
-    val: mir::ConstantKind<'tcx>,
-) -> InterpResult<'tcx, mir::DestructuredConstant<'tcx>> {
+    val: ConstValue<'tcx>,
+    ty: Ty<'tcx>,
+) -> Option<mir::DestructuredConstant<'tcx>> {
```

Comment:
So this now takes a ConstValue but returns a list of [ConstantKind](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/mir/enum.ConstantKind.html). Why the asymmetry? Almost seems better to still take a ConstantKind as argument like before?

Reply:
I looked at https://github.com/rust-lang/rust/pull/113291/files, not the individual commits... but I didn't scroll down far enough. :facepalm: sorry!